pkg/agent/watcher.go

```diff
@@ -47,7 +41,14 @@ func (w *Watcher) Start() {
 		log.Fatal(err)
 	}
 	defer watcher.Close()
-
+	if err = watcher.Add(w.configDir); err != nil {
+		log.Fatal(err)
+	}
+	// Add a first create event to the channel to force initial sync
```

Comment:
Is the intention of this initial event to replace ```syncer.Start()``` in main.go?

Reply:
Not exactly, the syncer logic still runs first but moved into `NewWatcher` function above since I thought it kind of made sense to be part of the watcher. It just populates the initial state of the `modelTracker` map.

Injecting the event here just forces an intial sync of the map with the configmap. It's done after the fsnotify watch is started to avoid missing any changes, otherwise in theory at least we could sync the configmap, then it changes, then the watch starts ... and so we've missed that change in between.

Oh hmm though I just realized that this might not work if the Events chan is not buffered, I guess it may hang here if so. But that should be fixable by just moving this to right after the goroutine below is started.

Reply:
Im confused by this events add... I though we should be looping through the modelDir upon start, not the `..data`. The reason is that: maybe the configmap hasn't been mounted yet, but the modelDir is still filled up.

Reply:
I didn't think there would necessarily be such an event. The watched files (symlink) will be there from the outset and so I don't _think_ will necessarily produce any create/modification fsnotify events.

Reply:
I don't think I tested with fsnotify, but I had experimented with a different inotify-based lib where from what I recall there wasn't an event to reflect the initial state. That's cool though, if it's not needed then this line can be removed. I'll try to also test when I get a chance

Reply:
yeah before a lgtm, I will run this in a docker container and do a test. lmk when you have tested so I can do a follow-up as well